tests/test_self.py

```diff
@@ -1321,11 +1281,190 @@ def test_regression_recursive_current_dir(self):
+
+    @staticmethod
+    def test_list_conf_levels() -> None:
+        """Test the --list-conf-levels flag."""
+        process = subprocess.run(
+            [
+                "pylint",
+                "--list-conf-levels",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert "Confidence(name='HIGH', description=" in process.stdout.decode("utf-8")
+
+    @staticmethod
+    def test_list_extensions() -> None:
+        """Test the --list-extensions flag."""
+        process = subprocess.run(
+            [
+                "pylint",
+                "--list-extensions",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert "pylint.extensions.empty_comment" in process.stdout.decode("utf-8")
+
+    @staticmethod
+    def test_full_documentation() -> None:
+        """Test the --full-documentation flag."""
+        process = subprocess.run(
+            [
+                "pylint",
+                "--full-documentation",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert "Pylint global options and switches" in process.stdout.decode("utf-8")
+
+    @staticmethod
+    def test_generate_rcfile() -> None:
+        """Test the --generate-rcfile flag."""
+        process = subprocess.run(
+            [
+                "pylint",
+                "--generate-rcfile",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        output = process.stdout.decode("utf-8")
+        assert "[MASTER]" in output
+        assert "profile" not in output
+
+        process_two = subprocess.run(
+            [
+                "pylint",
+                "--generate-rcfile",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert output == process_two.stdout.decode("utf-8")
+
+    @staticmethod
+    def test_long_help() -> None:
+        """Test the --long_help flag."""
+        process = subprocess.run(
+            [
+                "pylint",
+                "--long-help",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert "Environment variables:" in process.stdout.decode("utf-8")
+
+    @staticmethod
+    def test_errors_only() -> None:
+        """Test the --errors-only flag."""
+        with pytest.raises(SystemExit):
+            run = Run(["--errors-only"])
+            assert run.linter._error_mode
+
+    @staticmethod
+    def test_verbose() -> None:
+        """Test the --verbose flag."""
+        with pytest.raises(SystemExit):
+            run = Run(["--verbose"])
+            assert run.verbose
+
```

Comment:
I don't know why I can't make suggestion here, but we can parametrize most of the test using:

```python
    @pytest.mark.parametrize("options,expecteds", [
        [["--full-documentation"], ["Pylint global options and switches"]]
        # etc.
    ])
    def test_pylint_option(options,expecteds) -> None:
        process = subprocess.run(
            ["pylint"] + options,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            check=False,
        )
       output =  process.stdout.decode("utf-8")
        for expected in expecteds:
            assert expected in output
```

Reply:
Does that make the tests more readable in your opinion? I think the current tests are a bit clearer. They also allow easy expanding such as `test_help_msg` does.

Reply:
I'm not advocating to put all the options here, only the easy one where we check the output and that it's not outright crashing. We do need to copy paste for test that need to be extended.

carlijoy made some good looking parametrize here https://github.com/PyCQA/pylint/blob/main/tests/checkers/unittest_unicode/unittest_invalid_encoding.py#L91 that could make the list look better.

Reply:
Done!